timex/timex.go

```diff
@@ -0,0 +1,57 @@
+// Package timex should be called Rolex.
+package timex
+
+import (
+	"context"
+	"time"
+)
+
+// Ticker ...
+type Ticker struct {
+	C <-chan time.Time
+
+	tt   *time.Ticker
+	done chan bool
+	c    chan time.Time
+}
+
+// NewTicker ...
+func NewTicker(d time.Duration) *Ticker {
+	c := make(chan time.Time, 1)
```

Comment:
I'm not sure why this is a buffered channel. By doing so, you break the underlying ticker that drops ticks if the receiver is not ready (I think).

Reply:
I copied it from the underlying ticker...

Reply:
can't say i'd understand why they put it in. removed.

Reply:
The underlying implementation uses a 1 element buffer so that if the receiver is not ready, the timestamp sent on the channel is still correct. If the receiver is ready before the next tick, then all is good and it keeps firing on schedule even if the receiver is never quite ready when the ticker fires. But if it receiver is still not ready when the next tick fires, that is dropped. To make this work, you need a single element buffer.

However, because you are just passing through the ticks, you don't need this buffer. The underlying implementations buffer is sufficient to keep that same behaviour.

Reply:
👍